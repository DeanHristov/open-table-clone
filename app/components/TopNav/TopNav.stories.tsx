import * as React from "react";
import { Meta, Story } from "@storybook/react";
import TopNav, { ITopNavProps } from "./TopNav";

export default {
  title: "Components/TopNav",
  component: TopNav,
} as Meta;

export const Default: Story<ITopNavProps> = (args) => <TopNav {...args} />;

Default.args = {
  links: [
    { title: "For Businesses", url: "" },
    { title: "Mobile", url: "" },
    { title: "FAQ", url: "" },
  ],
};
