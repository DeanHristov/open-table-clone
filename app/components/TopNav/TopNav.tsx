"use client";

import { FC } from "react";
import Link from "next/link";

export interface ITopNavLink {
  title: string;
  url: string;
}

export interface ITopNavProps {
  links: ITopNavLink[];
}

const TopNav: FC<ITopNavProps> = ({ links }) => {
  return (
    <nav className="bg-gray-200 p-2 flex justify-end">
      <ul className="flex">
        {links.map((item, idx) => (
          <li
            key={`top-nav-item-${idx}`}
            className="text-xs cursor-pointer px-2 m-0"
          >
            <Link href={item.url}>{item.title}</Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default TopNav;
