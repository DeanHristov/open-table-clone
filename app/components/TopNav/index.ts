import TopNav, { ITopNavLink, ITopNavProps } from "./TopNav";

export type { ITopNavProps, ITopNavLink };
export default TopNav;
