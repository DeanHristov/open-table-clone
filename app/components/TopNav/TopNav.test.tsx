import { render } from "@testing-library/react";
import TopNav from "../TopNav";

describe("Component: <TopNav  {...} />", () => {
  it("Should be able to render default state", () => {
    const { container } = render(
      <TopNav
        links={[
          { title: "For Businesses", url: "" },
          { title: "Mobile", url: "" },
          { title: "FAQ", url: "" },
        ]}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
