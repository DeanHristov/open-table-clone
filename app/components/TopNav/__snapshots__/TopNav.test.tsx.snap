// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TopNav  {...} /> Should be able to render default state 1`] = `
<div>
  <nav
    class="bg-gray-200 p-2 flex justify-end"
  >
    <ul
      class="flex"
    >
      <li
        class="text-xs cursor-pointer px-2 m-0"
      >
        <a
          href=""
        >
          For Businesses
        </a>
      </li>
      <li
        class="text-xs cursor-pointer px-2 m-0"
      >
        <a
          href=""
        >
          Mobile
        </a>
      </li>
      <li
        class="text-xs cursor-pointer px-2 m-0"
      >
        <a
          href=""
        >
          FAQ
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
